--- redo_prebinding.c.orig	2009-09-21 22:10:31.000000000 -0500
+++ redo_prebinding.c	2009-09-21 22:28:01.000000000 -0500
@@ -82,7 +82,7 @@
  */
 #include <mach-o/redo_prebinding.h>
 #endif /* defined(LIBRARY_API) */
-
+#include <config.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -9774,7 +9774,7 @@
 	errors++;
 }
 #endif /* defined(LIBRARY_API) */
-
+#if HAVE_GETATTRLIST
 #include <sys/attr.h>
 /*
  * Structure defining what's returned from getattrlist.  It returns all the
@@ -9859,3 +9859,13 @@
 
 	return(FALSE);
 }
+#else
+static
+enum bool
+has_resource_fork(
+                char *filename)
+{
+  return FALSE;
+}
+#endif
+
