--- bytesex.h.orig	2006-09-17 13:54:31.000000000 -0700
+++ bytesex.h	2006-09-17 13:55:07.000000000 -0700
@@ -227,7 +227,11 @@
 /* current i386 thread states */
 #if i386_THREAD_STATE == 1
 extern void swap_i386_float_state(
-    struct __darwin_i386_float_state *fpu,
+#ifdef _STRUCT_X86_FLOAT_STATE32
+    _STRUCT_X86_FLOAT_STATE32 *fpu,
+#else
+    struct i386_float_state *fpu,
+#endif
     enum byte_sex target_byte_sex);
 
 extern void swap_i386_exception_state(
