--- ./arm.c.orig	2008-03-25 19:37:24.000000000 -0500
+++ ./arm.c	2009-06-11 12:30:39.000000000 -0500
@@ -16,6 +16,18 @@
 #include "opcode/arm.h"
 
 #define ISALNUM(xXx) (isalnum(xXx))
+#ifndef CPU_SUBTYPE_ARM_V5TEJ
+	#define CPU_SUBTYPE_ARM_V5TEJ		((cpu_subtype_t) 7)
+#endif
+#ifndef CPU_SUBTYPE_ARM_XSCALE
+	#define CPU_SUBTYPE_ARM_XSCALE		((cpu_subtype_t) 8)
+#endif
+#ifndef CPU_SUBTYPE_ARM_V7
+	#define CPU_SUBTYPE_ARM_V7			((cpu_subtype_t) 9)
+#endif
+#ifndef N_ARM_THUMB_DEF
+	#define N_ARM_THUMB_DEF	0x0008 
+#endif
 
 /*
  * These are the default cputype and cpusubtype for the arm architecture.
