--- /usr/include/CarbonCore/fp.h	2008-02-19 12:14:55.000000000 +0000
+++ /usr/include/CarbonCore/fp.h	2008-03-29 18:41:50.000000000 +0000
@@ -1314,7 +1314,7 @@
 *   dec2l       Similar to dec2num except a long is returned.                   *
 *                                                                               *
 ********************************************************************************/
-#if TARGET_CPU_PPC || TARGET_CPU_X86 || TARGET_CPU_PPC64 || TARGET_CPU_X86_64
+#if TARGET_CPU_PPC || TARGET_CPU_X86 || TARGET_CPU_PPC64 || TARGET_CPU_X86_64 || TARGET_CPU_ARM
     #define SIGDIGLEN      36  
 #endif
 #define      DECSTROUTLEN   80               /* max length for dec2str output */
