diff --git a/utmpx.h b/utmpx.h
index ee657d2..1279ae5 100644
--- a/utmpx.h
+++ b/utmpx.h
@@ -62,6 +62,7 @@
 
 #include <_types.h>
 #include <sys/time.h>
+#include <TargetConditionals.h>
 
 #ifndef _PID_T
 #define _PID_T
@@ -149,9 +150,11 @@ struct lastlogx *
 	getlastlogx(uid_t, struct lastlogx *);
 struct lastlogx *
 	getlastlogxbyname(const char*, struct lastlogx *);
+#if !TARGET_OS_IPHONE
 struct utmp;	/* forward reference */
 void	getutmp(const struct utmpx *, struct utmp *);
 void	getutmpx(const struct utmp *, struct utmpx *);
+#endif
 #endif /* !_POSIX_C_SOURCE || _DARWIN_C_SOURCE */
 
 struct utmpx *
