--- ./macosx_deployment_target.c.orig	2008-02-11 13:07:14.000000000 -0600
+++ ./macosx_deployment_target.c	2009-06-11 12:31:10.000000000 -0500
@@ -146,10 +146,10 @@
 	 * As a last resort we set the default to the highest known shipping
 	 * system to date.
 	 */
-	value->major = 6;
+	value->major = 5;
 	value->minor = 0;
-	value->name = allocate(strlen("10.6") + 1);
-	strcpy(value->name, "10.6");
+	value->name = allocate(strlen("10.5") + 1);
+	strcpy(value->name, "10.5");
 	warning("unknown value returned by sysctl() for kern.osrelease: %s "
 		"ignored (using %s)", osversion, value->name);
 	/* fall through to also warn about a possble bad user value */
