diff -ru uuid-1.6.0/Makefile.in uuid-1.6.0+iPhone/Makefile.in
--- uuid-1.6.0/Makefile.in	2007-05-19 19:58:11.000000000 +0000
+++ uuid-1.6.0+iPhone/Makefile.in	2008-01-02 02:41:23.000000000 +0000
@@ -253,7 +253,7 @@
 	-@if [ ".$(WITH_CXX)" = .yes ]; then \
 	    $(LIBTOOL) --mode=install $(SHTOOL) install -c -m 644 $(CXX_NAME) $(DESTDIR)$(libdir)/; \
 	fi
-	@$(LIBTOOL) --mode=install $(SHTOOL) install -c -s -m 755 uuid $(DESTDIR)$(bindir)/
+	@$(LIBTOOL) --mode=install $(SHTOOL) install -c -m 755 uuid $(DESTDIR)$(bindir)/
 	$(SHTOOL) install -c -m 644 $(S)/uuid.1 $(DESTDIR)$(mandir)/man1/
 	-@if [ ".$(WITH_PERL)" = .yes ]; then \
 	    (cd $(S)/perl && $(MAKE) $(MFLAGS) install DESTDIR=$(DESTDIR)); \
